gateway/src/shard/processor/socket_forwarder.rs

```diff
@@ -41,18 +37,12 @@ impl SocketForwarder {
         #[cfg(feature = "tracing")]
         tracing::debug!("starting driving loop");
 
-        loop {
-            let timeout = sleep(Self::TIMEOUT).fuse();
-            tokio::pin!(timeout);
-
-            let rx = Box::pin(self.rx.recv().fuse());
-            let tx = Box::pin(self.stream.next().fuse());
+        let timeout = sleep(Self::TIMEOUT);
+        tokio::pin!(timeout);
 
-            let select_message = future::select(rx, tx);
-
-            match future::select(select_message, timeout).await {
-                // `rx` future finished first.
-                Either::Left((Either::Left((maybe_msg, _)), _)) => {
+        loop {
+            tokio::select! {
```

Comment:
Actually, we shouldn't use `tokio::select!`, since it required us to manually specify a recursion limit. https://github.com/twilight-rs/twilight/pull/826

Reply:
My understanding from reading rust-lang/futures-rs#1917 is that's no longer the case as of Rust 1.45 which deprecated the need for `proc-macro-hack` (not in tokio's dependency tree).